### ISSUE TITLE
Env: Access denied for user 'username_here'@'X.X.X.X' error running wp-env start. 

### ISSUE THREAD

**Issue by StevenDufresne** [WordPress/gutenberg#29752]
## Description
When trying to start a new environment using `wp-env`, the CLI issues the following error: 
```
mysqlcheck: Got error: 1045: Access denied for user 'username_here'@'172.18.0.5' (using password: YES) when trying to connect
```

This issue is not specific to working locally using this repository, but exists wherever `@wordpress/env` is used.

## Step-by-step reproduction instructions
1. Destroy your local environment `npm run wp-env destroy`
2. Re-provision:  `npm run wp-env start`

Expect to see the errors. Should you not see the error, make sure your docker images are updated to the latest.

## Quick Debugging
- `username_here` was implemented in the WordPress docker image fairly recently (https://github.com/docker-library/wordpress/commit/8f5bcc1c980fce198bd191cfb7f51a3e76bb59cd#diff-7f28a0f62019f2d8b061638e8c3aea207404c1499f1c9241db3216fadd8f4530)
- the `wp-config.php` in the WordPress container matches the default config in the PR listed above.
- the `wp-config.php` includes the `getenv_docker` function that looks for a local `env_FILE` that does not exist in the WordPress container.
- When looking at the container using the docker desktop app, I see ``` No 'wp-config.php' found in /var/www/html, but 'WORDPRESS_...' variables supplied; copying '/usr/src/wordpress/wp-config-docker.php' (WORDPRESS_DB_NAME)```

## Environment Info

**Relevant NPM packages**
```
├─┬ @wordpress/env@3.0.2
│ ├── chalk@4.0.0 deduped
│ ├── copy-dir@1.3.0
│ ├── docker-compose@0.22.2
│ ├── extract-zip@1.6.7
│ ├── got@10.7.0
│ ├── inquirer@7.1.0 deduped
│ ├── js-yaml@3.13.1
│ ├── ora@4.0.2
│ ├── rimraf@3.0.2 deduped
│ ├── simple-git@2.35.0
│ ├── terminal-link@2.0.0
│ └── yargs@14.0.0
```

## WordPress information
trunk@https://github.com/WordPress/gutenberg/commit/647a41cbab8e44e5ecd79af381f54cec080a3758

**Comment by talldan** [comment]
What I don't know is how wp-env previously defined the DB username and password values. (cc @noahtallen, @noisysocks).

Struggling to find a reference to anything like that. Not sure if the docker container is supposed to work without these specified. If so, it may be an upstream issue.

**Comment by StevenDufresne** [comment]
Also noting that the feature added in the WordPress image was `un-beta`ed yesterday.

https://github.com/docker-library/wordpress/commit/fb920c697693bad49582c3a0c8bf1fa0e7642d2b

**Comment by noisysocks** [comment]
> What I don't know is how wp-env previously defined the DB username and password values. (cc @noahtallen, @noisysocks).

IIRC these were omitted because the `wordpress` image defaulted to `user = 'root'`, `password = ''` which is what the `mysql` image also defaults to.

**Comment by youknowriad** [comment]
Tried setting explicitely the environoment variables for user and password to "root" and "" in the docker-compose.yml generated by wp-env and for some reason I didn't manage to make it work

**Comment by pierlon** [comment]
The `wordpress` image was previously [creating `WORDPRESS_DB_NAME`](https://github.com/docker-library/wordpress/pull/572/files?file-filters%5B%5D=.php&file-filters%5B%5D=.sh&file-filters%5B%5D=.template&file-filters%5B%5D=No+extension&hide-deleted-files=true#diff-79738685a656fe6b25061bb14181442210b599f746faeaba408a2401de45038aL263) if it did not already exist.

**Comment by youknowriad** [comment]
Thanks  for the hint @pierlon this helped me figure out a potential fix here https://github.com/WordPress/gutenberg/pull/29770

**Comment by pierlon** [comment]
@youknowriad I've also created a PR that reintroduces the feature of creating the database on container startup: https://github.com/docker-library/wordpress/pull/575.

**Comment by noahtallen** [comment]
Resolved in #29800. This fix will need to be pushed to npm before anyone outside of gutenberg will see it, and I'm not sure when that will happen.

See https://github.com/WordPress/gutenberg/pull/29800#pullrequestreview-610413324 for possible follow-up tasks.
